### DISCUSSION TITLE
Scanning only the latest commits

### DISCUSSION THREAD

**Discussion by Yogesh-BK** [zricethezav/gitleaks#936] [Q&A]
Hi team, 
          We are planning to use gitleaks to scan the secrets in the our pipelines, Is there any way to configure the build to scan only the latest commits (i.e., commits that are present in the current build and also the commits of old build only if there are any secrets present on that build), Thanks in advance

**Comment by ttiot** [comment]
If someone car take this point, it would be nice! 
I am interested too

**Comment by weineran** [comment]
@Yogesh-BK @ttiot I'm not sure I fully understand your requirements, but you might want to check out gitleaks-action: https://github.com/gitleaks/gitleaks-action

This is GitHub Action wrapper for gitleaks. You can configure it to run automatically on every pull request and push, in which case it will only scan the new commits associated with that PR or that push.

Does that sounds like it does what you need?

**Reply by ttiot** [reply]
Thanks @weineran.
I will check if it fits my need. 
This is not for GitHub, I would like to use on gitlab. 

Thank you

**Comment by ttiot** [comment]
Hello.

I finally manage to add a pre-receive hook on my gitlab instance.
Thank you

**Comment by roshvin** [comment]
hi @ttiot could you pls share some inside or the script used to make it as pre-receive hook.
